File: hash/lonely_integers/lonely_integers.py

def find_lonely_integers(nums: list[int]) -> list[int]:  # LeetCode Q.2150.
    nums2counts: dict[int, int] = dict()
    for num in nums:
        if num not in nums2counts.keys():
            nums2counts.update({num: 0})
        nums2counts[num] += 1
    
    lonely_nums = []
    for num in nums:
        if nums2counts[num] == 1:
            if num - 1 not in nums2counts.keys():
                if num + 1 not in nums2counts.keys():
                    lonely_nums.append(num)
    
    return lonely_nums

File: hash/lonely_integers/lonely_integers.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var find_lonely_integers = function (nums) { // LeetCode Q.2150.
    let nums2counts = new Map();
    for (num of nums) {
        if (!nums2counts.has(num))
            nums2counts.set(num, 0);
        nums2counts.set(num, nums2counts.get(num) + 1);

    }
    let lonely_nums = [];
    for (num of nums) {
        if (nums2counts.get(num) == 1) {
            if (!nums2counts.has(num - 1) && !nums2counts.has(num + 1))
                lonely_nums.push(num);
        }
    }

    return lonely_nums;
};